### PR TITLE
scripts: hid_configurator: GUI DFU handling update

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds displaying firmware version from file and reboot with image swap handling.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@ppryga, could you review ?

File: scripts/hid_configurator/gui/gui.py [hunk 1]
```diff
@@ -123,14 +142,25 @@ def show_load_list(self):
     def load_list(self, files_path):
         print(files_path)
         self.filepath = files_path[0]
-        self.root.ids.dfu_buttons.ids.choose_file_button.disabled = True
         self.root.ids.dfu_buttons.ids.start_uploading_button.disabled = False
+
+        if '\\' in self.filepath:
```
Comment (ppryga) [inline scripts/hid_configurator/gui/gui.py]: Is it needed? Why don't you use os.path?

File: scripts/hid_configurator/gui/gui.py [hunk 2]
```diff
@@ -142,13 +172,31 @@ def dfu(self, update_progressbar):
...
+
         dfu_label.text = info
         notification.notify(app_name='GUI Configurator', app_icon='nordic.ico', title=info, message=message)
 
-    def perform_fwreboot(self):
-        print('Performing fwreboot, please wait')
-        self.device.perform_fwreboot()
+    def dfu_fwreboot(self, update_reboot_animation):
+        dfu_label = self.root.ids.dfu_label
+        progression_label = self.root.ids.dfu_buttons.ids.progression_label
```
Comment (ppryga) [inline scripts/hid_configurator/gui/gui.py]: created variable just to make single assignment at the end of the function. Please use self.root.ids.dfu_buttons.ids.progression_label.

File: scripts/hid_configurator/gui/gui.py [hunk 3]
```diff
@@ -142,13 +172,31 @@ def dfu(self, update_progressbar):
...
+        dfu_label.text = 'Image swap in progress'
```
Comment (ppryga) [inline scripts/hid_configurator/gui/gui.py]: Is it visible at all? Before end of the function dfu_label.text is overwritten.

File: scripts/hid_configurator/gui/gui.py (hunk 1 above)
Comment (MarekPieta) [reply scripts/hid_configurator/gui/gui.py]: Thanks for suggestion. Changed.

File: scripts/hid_configurator/gui/gui.py (hunk 3 above)
Comment (MarekPieta) [reply scripts/hid_configurator/gui/gui.py]: Yes, `device.perform_dfu_fwreboot(update_reboot_animation)` takes a while (thread waits for device to complete firmware swap).

File: scripts/hid_configurator/gui/gui.py (hunk 2 above)
Comment (MarekPieta) [reply scripts/hid_configurator/gui/gui.py]: done

**Comment by MarekPieta** [comment]
@pdunaj, could you review?

**Pull request merged**